packages/wonder-blocks-timing/src/hooks/__tests__/use-interval.test.ts

```diff
@@ -1,123 +1,520 @@
         // Arrange
-        const action = jest.fn();
+        const action1 = jest.fn();
+        const action2 = jest.fn();
         const {rerender} = renderHook(
-            ({active}: any) => useInterval(action, 500, active),
-            {initialProps: {active: true}},
+            ({action}: any) => useInterval(action, 500),
+            {
+                initialProps: {action: action1},
+            },
         );
 
         // Act
-        rerender({active: false});
+        rerender({action: action2});
         jest.advanceTimersByTime(501);
 
         // Assert
-        expect(action).not.toHaveBeenCalled();
+        expect(action2).toHaveBeenCalledTimes(1);
     });
 
-    it("should reset the interval if 'intervalMs' is changes", () => {
+    it("should only call setInterval once even if action changes", () => {
         // Arrange
-        const action = jest.fn();
-
-        // Act
+        const intervalSpy = jest.spyOn(global, "setInterval");
+        const action1 = jest.fn();
+        const action2 = jest.fn();
         const {rerender} = renderHook(
-            ({timeoutMs}: any) => useInterval(action, timeoutMs, true),
-            {initialProps: {timeoutMs: 500}},
+            ({action}: any) => useInterval(action, 500),
+            {
+                initialProps: {action: action1},
+            },
         );
-        rerender({timeoutMs: 1000});
-        jest.advanceTimersByTime(501);
+
+        // Act
+        rerender({action: action2});
 
         // Assert
-        expect(action).not.toHaveBeenCalled();
-        jest.advanceTimersByTime(1001);
-        expect(action).toHaveBeenCalled();
+        expect(intervalSpy).toHaveBeenCalledTimes(1);
     });
 
-    it("should not reset the interval if 'action' changes", () => {
+    it("should not reset the interval if the action changes", () => {
         // Arrange
         const action1 = jest.fn();
         const action2 = jest.fn();
-        const timeoutSpy = jest.spyOn(window, "setTimeout");
-
-        // Act
         const {rerender} = renderHook(
-            ({action}: any) => useInterval(action, 500, true),
-            {initialProps: {action: action1}},
+            ({action}: any) => useInterval(action, 500),
+            {
+                initialProps: {action: action1},
+            },
         );
-        // NOTE: For some reason setTimeout is called twice by the time we get
-        // here.  I've verified that it only gets called once inside the hook
-        // so something else must be calling it.
-        const callCount = timeoutSpy.mock.calls.length;
+
+        // Act
+        jest.advanceTimersByTime(250);
         rerender({action: action2});
-        jest.advanceTimersByTime(501);
+        jest.advanceTimersByTime(751);
 
         // Assert
-        expect(timeoutSpy).toHaveBeenCalledTimes(callCount);
+        expect(action1).not.toHaveBeenCalled();
+        expect(action2).toHaveBeenCalledTimes(2);
```

Comment:
Does this mean that `action2` is called immediately when the hook is re-rendered?  It seems a little weird to me, but also I can't make a strong argument as to why any other default behaviour would be preferable and people can change the reset behaviour anyways.  Also, I'm guessing that changing the `action` is not something that's a very common use case.

Reply:
No, the action is not called when anything renders. It is called when the interval fires. This test shows that changing the action means it will get called when the next interval occurs - it won't reset the interval.